src/ngrx/1-why-ngrx/why-ngrx.md

```diff
@@ -5,6 +5,103 @@
+
+    PRO TIP: if you use [@bitovi/eslint-config](https://github.com/bitovi/eslint-config), there’s no need to worry: NgRx recommended ESLint rules will be configured for you.
+
+3. Maintainability
+
+    While NgRx allegedly has a lot of boilerplate code, NgRx does help keep your code tidy!
+
+4. Performance
+
+    NgRx is carefully implemented with performance in mind, and it complements the Angular Ecosystem perfectly because of its Reactive approach.
+
+## Key Concepts
+
+Understand the key concepts of NgRx
+
+### Redux Pattern
+
+The Redux pattern manages the global state of an application and aims at code testability and predictability.
+
+Redux is known for its boilerplate code. At first, it might seem like a lot of effort to implement Redux, but in fact it is a tradeoff. Redux adds a layer of indirection in order to improve maintainability, testability and predictability of your application. The tradeoff pays off more often than not particularly if your application is complex, large and maintained by a large team.
+
+[Redux expects updates to the state to be done immutably](https://redux.js.org/tutorials/essentials/part-1-overview-concepts#immutability). Not only Immutability helps Redux implement patterns that help testability and predictability, it also improves performance, as change detection is achieved with less effort.
+
+#### Store
+
+A Store is an object that wraps the state. The Store also provides methods to update the state and listen to state changes.
+
+#### Actions
+
+Actions are objects that represent individual events in your application.
+
+Actions have two properties:
+
+- `type` describes the event;
+- `payload` (optional) holds additional information.
+
+Actions are necessary to update the state.
+
+#### Reducers
+
+To put it simply, reducers are functions that take the state and an action as arguments, and return a new state. In other words, Reducers are functions that, given an action, know how to update the state.
+
+Reducers functions must be `pure`:
+
+- do not depend on any information other than the action and current state;
+- do not perform side-effects, like mutating the current state.
+
+#### Dispatch
+
+A Dispatch is a Store method that will provide an action to the Reducer function. You can see the Dispatch method as the API to update the state.
+
+#### Selectors
+
+Selectors are functions that extract data from the state. Selectors can return derived data or pieces of the state.
+
+#### One-way Data Flow
+
+To help you understand how Redux pieces work together, let’s take a look at the most basic representation of how data is passed around in an application that uses Redux, also known as a _one-way data flow_, [taken from the Redux Official Documentation](https://redux.js.org/tutorials/fundamentals/part-2-concepts-data-flow#state-management).
+
+<figure>
+    <img src="../static/img/ngrx/1-why-ngrx/ngrx-one-way-data-flow.png" alt="One-way data flow">
```

Comment:
image looks broken on my end locally. Is that normal?

Reply:
it could be that locally, and in production, they behave differently.

Reply:
you'll have to run that `rebuild-assets` command to get images to show up when running locally